### PR TITLE
YUNIKORN-798 the REST APIs of getting queue application returns "Queu…

### PR DESCRIPTION
**Pull request by chia7712**
…e not found" even though the queue is existent

### What is this PR for?
the queue having no apps is considered nonexistent.
 
```
 	if !appExistsInQueue {
		buildJSONErrorResponse(w, "Queue not found", http.StatusBadRequest)
		return
	}
```

### What type of PR is it?
* [x] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-798

### How should this be tested?
new unit test

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/webservice/handlers.go [hunk 1]
```diff
@@ -687,28 +687,25 @@ func getQueueApplications(w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
-	var appExistsInPartition = false
-	var appExistsInQueue = false
-	var appsDao []*dao.ApplicationDAOInfo
+
 	partitionContext := schedulerContext.GetPartitionWithoutClusterID(partition)
-	if partitionContext != nil {
-		appExistsInPartition = true
-		appList := partitionContext.GetApplications()
```
Comment (chia7712) [inline pkg/webservice/handlers.go:696]: this call copy all applications of whole partition. this PR optimizes it by copying apps from only specify queue.

**Review by manirajv06 — COMMENTED** [review]
Yes, "Queue not found" error message is misleading when queue doesn't have any applications. It should go something like "No applications.." etc.

Yes, optimisation is good and main outcome of this JIRA. Some minor comments..

**Comment by chia7712** [comment]
> It should go something like "No applications.." etc.

Did you mean it should return error response if the queue has no applications? If so, it seems to me returning empty array is better way. It is normal that a queue has no application. Hence, the empty array is more suitable to be the response of such REST apis.

**Review by manirajv06 — APPROVED** [review]
Yes, returning [] as successful response makes sense. LGTM.

**Pull request merged**